File: app/src/generic/java/org/tasks/billing/BillingClientImpl.java
package org.tasks.billing;

import android.app.Activity;
import android.content.Context;
import androidx.annotation.Nullable;
import org.tasks.analytics.Tracker;

public class BillingClientImpl implements BillingClient {

  public static final String TYPE_SUBS = "";

  public BillingClientImpl(Context context, Inventory inventory, Tracker tracker) {}

  @Override
  public void queryPurchases() {}

  @Override
  public void initiatePurchaseFlow(
      Activity activity, String sku, String skuType, @Nullable String oldSku) {}

  @Override
  public void addPurchaseCallback(OnPurchasesUpdated onPurchasesUpdated) {}

  @Override
  public void consume(String sku) {}
}

File: app/src/generic/java/org/tasks/billing/BillingClientImpl.kt
package org.tasks.billing

import android.app.Activity
import android.content.Context
import org.tasks.analytics.Firebase
import org.tasks.jobs.WorkManager

@Suppress("UNUSED_PARAMETER")
class BillingClientImpl(
    context: Context,
    inventory: Inventory,
    firebase: Firebase,
    workManager: WorkManager
) : BillingClient {
    override suspend fun queryPurchases(throwError: Boolean) {}
    override suspend fun initiatePurchaseFlow(
            activity: Activity,
            sku: String,
            skuType: String,
            oldPurchase: Purchase?
    ) {}

    override suspend fun acknowledge(purchase: Purchase) {}
    override suspend fun getSkus(skus: List<String>): List<Sku> = emptyList()

    override suspend fun consume(sku: String) {}

    companion object {
        const val TYPE_SUBS = ""
    }
}